server/game/cards/events/01/tearsoflys.js

```diff
@@ -5,32 +5,22 @@ class TearsOfLys extends DrawCard {
         this.reaction({
             max: ability.limit.perChallenge(1),
             when: {
-                afterChallenge: ({challenge}) => (
-                    challenge.attackingPlayer === this.controller &&
-                    challenge.winner === this.controller &&
-                    challenge.challengeType === 'intrigue'
-                )
+                afterChallenge: event => event.challenge.attackingPlayer === this.controller && event.challenge.winner === this.controller &&
+                                         event.challenge.challengeType === 'intrigue'
             },
-            handler: () => {
```

Comment:
I've discussed this with @ystros not too long ago. Using the target API, even when the card technically doesn't target has two advantages: cleaner looking card implementations and not getting prompted when the card effect does not have the chance to change the game state. A disadvantage could arise when something like High Septon gets implemented. Even so, that might be resolved by adding a flag to target, something like trueTarget: false in this case.